lib/src/main/java/com/diffplug/spotless/generic/LicenseHeaderStep.java

```diff
@@ -323,12 +326,13 @@ private String addOrUpdateLicenseHeader(String raw) {
 								return raw;
```

Comment:
It looks suspicious to me `replaceFileName` is not applied here.

It would be easier to read if existing code where pushed into a `replaceYear`. Then `replaceYear` and `replaceFileName` could be applied independently the one to the other(s) (and future possible placeholder s(like https://github.com/diffplug/spotless/issues/323)).

Reply:
> It looks suspicious to me replaceFileName is not applied here.

Fixed 318f453

> It would be easier to read if existing code where pushed into a replaceYear. Then replaceYear and replaceFileName could be applied independently the one to the other(s) (and future possible placeholder s(like https://github.com/diffplug/spotless/issues/323)).

I agree with this. However, I think this refactoring is hard and takes more time for me.

Reply:
I also feel the following is suspicious:

	if (contentMatcher.start() == yearSepOrFull.length() && raw.startsWith(yearSepOrFull)) {
		// if no change is required, return the raw string without
		// creating any other new strings for maximum performance
		return raw;
	}

I'm not sure the refactoring is this hard if you get back to the original code, wrap the existing logic in `replaceYear`, and immediately call `replaceFileName` after it. The potential bugs ans suspicions (you confirmed as correct) would spare time on your side.

Reply:
Please give me some time. I'll try it.

Reply:
Fixed d4d3814

Reply:
Thanks, it looks much leaner. LGTM.